source/macos/AdaptiveCards/AdaptiveCards/Views/ACRMultilineInputTextView.swift

```diff
@@ -7,10 +7,27 @@ class ACRMultilineInputTextView: NSView, NSTextViewDelegate {
     @IBOutlet var textView: ACRTextView!
     
     private var placeholderAttrString: NSAttributedString?
+    weak var errorMessageHandler: ErrorMessageHandlerDelegate?
     private let config: RenderConfig
     private let inputConfig: InputFieldConfig
-    var maxLen: Int = 0
+    var maxLen = 0
     var id: String?
+    var regex: String? {
+        didSet {
+            // handling case when regex is not supplied and is set to "". Replacing it with ".*"
+            guard let currValue = regex else { return }
+            regex = currValue.isEmpty ? ".*" : currValue
+        }
+    }
+    var isRequired = false
+    var textFieldShowsError = false
+    var hasMouseInField = false
+    var hasError: Bool {
+        get {
+            // if string value is empty, then check if it is required. In case string has value, check if its valid regex
+            return textView.string.isEmpty ? isRequired : textView.string.range(of: regex ?? ".*", options: .regularExpression, range: nil, locale: nil) == nil
+        }
+    }
```

Comment:
Good that you centralized the common validation logic!
But I feel a better name can be used here, instead of `hasError`.

Also, get keyword can be omitted.

Reply:
I have made a commit with the refined name and it is there in ```ACRInputValidator.swift``` file